test/module/shared_model/validators/transaction_validator_test.cpp

```diff
@@ -126,3 +126,74 @@ TEST_F(TransactionValidatorTest, StatelessInvalidTest) {
+  unsigned long long gap =
+      std::chrono::minutes(3) / std::chrono::milliseconds(1);
+  iroha::protocol::Transaction tx = generateEmptyTransaction();
+  tx.mutable_payload()->set_created_time(created_time + gap);
+  auto payload = tx.mutable_payload();
+
+  // Iterate through all command types, filling command fields with valid values
+  iterateContainer([] { return iroha::protocol::Command::descriptor(); },
+                   [&](auto field) {
+                     // Add new command to transaction
+                     auto command = payload->add_commands();
+                     // Set concrete type for new command
+                     return command->GetReflection()->MutableMessage(command,
+                                                                     field);
+                   },
+                   [this](auto field, auto command) {
+                     // Will throw key exception in case new field is added
+                     field_setters.at(field->name())(
+                         command->GetReflection(), command, field);
+                   },
+                   [] {});
+
+  shared_model::validation::DefaultTransactionValidator transaction_validator;
+  auto answer = transaction_validator.validate(
+      detail::makePolymorphic<proto::Transaction>(tx));
+
+  ASSERT_FALSE(answer.hasErrors()) << answer.reason();
+}
+
+/**
+ * @given transaction from far future(30 min) made of commands with valid fields
+ * @when commands validation is invoked
+ * @then answer has an error
+ */
+TEST_F(TransactionValidatorTest, StatelessInvalidFromFarFutureTest) {
+  unsigned long long gap =
+      std::chrono::minutes(30) / std::chrono::milliseconds(1);
+  iroha::protocol::Transaction tx = generateEmptyTransaction();
+  tx.mutable_payload()->set_created_time(created_time + gap);
+  auto payload = tx.mutable_payload();
+
+  // Iterate through all command types, filling command fields with valid values
+  iterateContainer([] { return iroha::protocol::Command::descriptor(); },
+                   [&](auto field) {
+                     // Add new command to transaction
+                     auto command = payload->add_commands();
+                     // Set concrete type for new command
+                     return command->GetReflection()->MutableMessage(command,
+                                                                     field);
+                   },
+                   [this](auto field, auto command) {
+                     // Will throw key exception in case new field is added
+                     field_setters.at(field->name())(
+                         command->GetReflection(), command, field);
+                   },
+                   [] {});
+
+  shared_model::validation::DefaultTransactionValidator transaction_validator;
+  auto answer = transaction_validator.validate(
+      detail::makePolymorphic<proto::Transaction>(tx));
+
+  ASSERT_TRUE(answer.hasErrors());
+}
```

Comment:
and here

Reply:
there is a problem with invalid test case in field validator. It is quite hard to make it with the current FieldTestCase because it requires to provide exact error message. But the field_validator gets current time and generates error message inside validateCreatedTime method. It is not possible to provide correct error message during creation of testcase because the test itself will be executed later.

Reply:
It is possible to make separate test, but it will be the exact copy of corresponding tests from transaction_validator_test.

Reply:
We can add compile-time parametrization of time functions such as iroha::time::now(), which will allow us to mock them and test error messages properly, but this might be a separate task.

Reply:
I've made a new flag in FieldTestCase in order to disable error message checking. Also, I've created new test case.